src/datatypes.ts

```diff
@@ -153,6 +156,110 @@ export function deserializeDatetime(literalString: string): Date | null {
   return date;
 }
 
+/**
+ * Time type for time data type attributes
+ */
+export type Time = {
+  hour: number;
+  minute: number;
+  second: number;
+  millisecond?: number;
+  timezoneHourOffset?: number;
+  timezoneMinuteOffset?: number;
+};
+
+/**
+ * @internal
+ * @param value Value to serialise.
+ * @returns String representation of `value` in UTC.
+ * @see https://www.w3.org/TR/xmlschema-2/#time-lexical-representation
+ */
+export function serializeTime(value: Time): string {
+  const utcHours =
+    value.hour + (value.timezoneHourOffset ? value.timezoneHourOffset : 0);
+  const utcMinutes =
+    value.minute +
+    (value.timezoneMinuteOffset ? value.timezoneMinuteOffset : 0);
+
+  let millisecondString;
+  // get millisecond into three digits
+  if (value.millisecond) {
+    if (value.millisecond < 10) {
+      millisecondString = "00" + value.millisecond;
+    } else if (value.millisecond < 100) {
+      millisecondString = "0" + value.millisecond;
+    } else {
+      millisecondString = value.millisecond;
+    }
+  }
+
+  return (
+    (utcHours < 10 ? "0" + utcHours : utcHours) +
+    ":" +
+    (utcMinutes < 10 ? "0" + utcMinutes : utcMinutes) +
+    ":" +
+    (value.second < 10 ? "0" + value.second : value.second) +
+    (value.millisecond ? "." + millisecondString + "Z" : "Z")
+  );
+}
+
+/**
+ * @internal
+ * @param literalString Value to deserialise.
+ * @returns Deserialized time, or null if the given value is not a valid serialised datetime.
+ * @see https://www.w3.org/TR/xmlschema-2/#time-lexical-representation
+ */
+export function deserializeTime(literalString: string): Time | null {
+  // Time in the format described at
+  // https://www.w3.org/TR/xmlschema-2/#time-lexical-representation
+  // (without constraints on the value).
```

Comment:
Hmm, so this is a leftover from the dateTime equivalent, which seems to accept e.g. `25:61` for the time part just fine. :shrug: 

```suggestion
```